### PR TITLE
Update Bulk with batchsize and multithreading

### PR DESCRIPTION
**Pull request by jon-wobken**
Re-implemented multiprocessing using multithreading with batchsizes.

### CODE REVIEW

File: CHANGES [hunk 1]
```diff
@@ -1,3 +1,8 @@
+v0.75.1
+Features
+-------
+- [345] Re-implented mutli-batch processing using multi-threading
```
Comment (jankatins) [inline CHANGES]: Typo: multi-batch

File: README.rst [hunk 2]
```diff
@@ -231,7 +231,7 @@ To retrieve a list of top level description of instance metadata, user:
 Using Bulk
 ----------
 
-You can use this library to access Bulk API functions. 
+You can use this library to access Bulk API functions. Batch sizes can be delcared using the batchsize arguement. Default is 10,000 records. Batch sizes are only supported in the following methods: insert,upsert,update,delete,hard_delete.
```
Comment (jankatins) [inline README.rst]: Typo: declared ... argument

File: simple_salesforce/bulk.py [hunk 3]
```diff
@@ -163,8 +165,36 @@ def _get_batch_results(self, job_id, batch_id, operation):
         return result.json()
 
     #pylint: disable=R0913
+    def worker(self, data, job, operation, wait=5):
+        """ Add batches to concurrent worker threads"""
+        result_list = []
+        while True:
+            data = data[0]
```
Comment (jankatins) [inline simple_salesforce/bulk.py]: A comment for why this unpacking is needed would be nice

File: README.rst (hunk 2 above)
Comment (jankatins) [reply README.rst]: Fun: We have been advised by salesforce to send our data in batches with size 75 (to work around apex problems).

Comment (jon-wobken) [reply README.rst]: Glad it can be helpful! 

Note to self: Spell check is your friend

File: simple_salesforce/bulk.py (hunk 3 above)
Comment (jon-wobken) [reply simple_salesforce/bulk.py]: Added additional comments on why unpacking was necessary and included a shutdown of the threads once the worker function has processed the batches.

**Comment by Phlogi** [comment]
Could this please be merged? I'd be willing to test.

**Comment by jon-wobken** [comment]
> Could this please be merged? I'd be willing to test.

I'll look to merge this over the weekend. I will need to make some updates for conflicts compared to current master branch.

**Comment by jon-wobken** [comment]
Updated Pull request with rebased master in #392

**Pull request closed without merging**